pandora/workers/pdf.py

```diff
@@ -0,0 +1,134 @@
+                        if js_type != ("null", "null"):
+                            if js_type[0] == "string":  # Directly embedded JavaScript
+                                js_code = self._truncate_string(js_type[1])
+                                js_scripts.append(js_code)
+                            elif js_type[0] == "xref":  # JavaScript referenced in another object
+                                js_ref = int(js_type[1].split()[0])
+                                js_code = doc.xref_stream(js_ref).decode('utf-8')
+                                js_code = self._truncate_string(js_code)
+                                js_scripts.append(js_code)
+
+        except Exception as e:
+            self.logger.warning(f'Unable to extract JavaScript from PDF file: {e}')
+
+        return js_scripts
+
+    def _detect_suspicious_objects(self, file_path: str) -> list:
+        try:
+            suspicious_objects = []
+            doc = fitz.open(file_path)
+            for xref in range(1, doc.xref_length()):
+                obj_dict = doc.xref_object(xref, compressed=True)
+                # Check for /AA and /OpenAction
+                for keyword in ["/AA", "/OpenAction"]:
+                    if keyword in obj_dict:
+                        action_content = self._truncate_string(obj_dict)
+                        suspicious_objects.append(f'{keyword} found in object {xref}: {action_content}')
+                        # Attempt to extract the object content if possible
+                        try:
+                            content = doc.xref_stream(xref)
+                            content = self._truncate_string(content.decode('utf-8', errors='ignore'))
+                            suspicious_objects.append(content)
+                        except:
+                            pass
+
+        except Exception as e:
+            self.logger.warning(f'Unable to detect suspicious objects in PDF file: {e}')
+
+        return suspicious_objects
+
+    def _detect_embedded_files(self, file_path: str) -> list:
+        try:
+            embedded_files = []
+            doc = fitz.open(file_path)
+            for item in range(doc.embfile_count()):
+                embedded_files.append(doc.embfile_info(item))
+
+        except Exception as e:
+            self.logger.warning(f'Unable to detect embedded files in PDF file: {e}')
+
+        return embedded_files
+
+    def _truncate_string(self, string: str, max_length: int = 100) -> str:
+        """Truncate a string to the max_length"""
+        if len(string) > max_length:
+            return string[:max_length] + "..."
+        return string
+
+
+    def analyse(self, task: Task, report: Report, manual_trigger: bool=False) -> None:
+        if not task.file.is_pdf:
+            report.status = Status.NOTAPPLICABLE
+            return
+
+        self.logger.debug(f'Analysing PDF file {task.file.path}...')
+        try:
+            is_encoded = ""
```

Comment:
Should be `is_encrypted`, right?

Reply:
You're right. I change the variable name.